### ISSUE TITLE
Add support for assertions

### ISSUE THREAD

**Issue by mindplay-dk** [jashkenas/coffeescript#2336]
Assertions are important in JS/DOM projects, because early failure often helps diagnose problems that would otherwise be very difficult to trace back to the root of the problem - due to the event-driven nature of JS/DOM code, often the call that caused an error is no longer on the call-stack, so assertions are pretty essential.

It would be nice to have an assert-statement in coffee, and basically it could just compile down to console.assert() calls - but with two important features: the ability to turn off code-generation for assertions (in production-ready scripts) and the ability to add a comment that gets preserved in the generated JS. (when assertions are switched on.)

So basically:

```
el = $('#foo')
assert el.length > 0 # element must be present
```

Compiles down to:

```
var el;
el = $('#foo');
console.assert(el.length > 0); // element must be present
```

Or with assertions switched off:

```
var el;
el = $('#foo');
```

Should be low-hanging fruit, I think? Nothing fancy, but a simple and valuable extension, I think.

**Comment by emptist** [comment]
how about this now?

**Comment by GeoffreyBooth** [comment]
This is already handled by other tools, I don’t think CoffeeScript needs to support it natively.

I’ve used `envify` in the past: https://github.com/hughsk/envify

To quote from their readme:

> envify will replace your environment variable checks with ordinary strings - only the variables you use will be included, so you don’t have to worry about, say, AWS_SECRET_KEY leaking through either. Take this example script:
> 
> ```
> if (process.env.NODE_ENV === "development") {
>   console.log('development only')
> }
> ```
> After running it through envify with NODE_ENV set to production, you’ll get this:
> 
> ```
> if ("production" === "development") {
>   console.log('development only')
> }
> ```
> By running this through a good minifier (e.g. UglifyJS2), the above code would be stripped out completely.

**Comment by emptist** [comment]
Now I'm trying to wrap simple-assert a little bit to get work done.
